### DISCUSSION TITLE
Returning deleted item content

### DISCUSSION THREAD

**Discussion by drnextgis** [stac-utils/pgstac#326] [Ideas]
While working on https://github.com/stac-utils/pgstac/issues/323, I considered adding support for a callback in the `delete_item` method https://github.com/stac-utils/stac-fastapi-pgstac/blob/main/stac_fastapi/pgstac/transactions.py#L179 that could be triggered on the deleted STAC item. However, I realized that in pgstac, the `delete_item` function https://github.com/stac-utils/pgstac/blob/6c569c0ee56bd9519ad947cbcaa8bfbcfade15bb/src/pgstac/sql/003a_items.sql#L292 is defined with `RETURNS VOID`, which means the content of the deleted item isn't available on the client side. The idea was to pass the item's content to a callback that could handle the removal of any underlying assets. Since `delete_item` in pgstac is always executed on a single item, it doesn't seem like a significant overhead to return the content of the removed item. What are your thoughts on this?

**Comment by vincentsarago** [comment]
Is that a `regular` pattern in API design that `delete` method return the deleted API?

Thinking about #323 I was maybe thinking about adding callback methods on all the `transactional` methods. While the Callback would be a SQL methods (so it could not delete the asset directly, it could maybe create a Table with all the assets to be deleted and then this table would be used by a Cron that would delete the asset) 🤷

I'm just expressing my personal opinion, maybe David will agree about returning the Item :-)

**Reply by bitner** [reply]
I actually originally did return the deleted item, but the spec (at least as it existed then) had the return as void. It'd definitely be super easy to return the item, but if it's against the spec, I'd be inclined to do it as a separate function.
